client/allocrunner/consul_grpc_sock_hook.go

```diff
@@ -184,55 +185,78 @@ func (p *grpcSocketProxy) run(alloc *structs.Allocation) error {
 	default:
 	}
 
-	// make sure either grpc or http consul address has been configured
-	if p.config.GRPCAddr == "" && p.config.Addr == "" {
-		return errors.New("consul address must be set on nomad client")
+	clusterNames := set.New[string](0)
```

Comment:
```suggestion
	clusterNames := set.New[string](1)
```

We probably expect to find at least one cluster right? 😅